### ISSUE TITLE
Files without tests missing from final report

### ISSUE THREAD

**Issue by dnaka91** [mozilla/grcov#617]
I noticed that files without any test data are not included in the grcov outputs (specifically the HTML report). This additionally seems to give a wrong coverage percentage, showing a higher percentage than expected as all files without coverage data are not counted in.

Noticed this in my Rust projects so it might be specific to Rust eventually.

Is there any option to tell grcov to take uncovered files into account when generating the report?
Couldn't find anything so far that would help in this regards.

**Comment by TilBlechschmidt** [comment]
I ran across the same issue. It is related to the fact that compilers are way too smart these days and strip all the unused function from the binary. In a test-scenario this means that all the files that have no used functions under a `#[cfg(test)]` scenario will not be included. This in turn, results in grcov having no clue about them even existing.

For more information, have a look at this Rust PR: https://github.com/rust-lang/rust/pull/31368 

TL;DR Run your code with the new compiler flag `-Clink-dead-code` like this

```bash
export RUSTFLAGS="-Zinstrument-coverage -Clink-dead-code"
export LLVM_PROFILE_FILE="coverage-pid%p.profraw"

cargo +nightly test --tests

# Run grcov now as usual
```

**Comment by taiki-e** [comment]
This is a bug of `-Zinstrument-coverage`: https://github.com/rust-lang/rust/issues/86177

Note that `-Clink-dead-code` works as a workaround, but `-Clink-dead-code` is basically considered incompatible with `-Zinstrument-coverage`. So it is likely to cause other problems. 
See also https://github.com/taiki-e/cargo-llvm-cov/pull/26

**Comment by snue** [comment]
https://github.com/rust-lang/rust/issues/86177 has been resolved.
I think this issue can be closed.
